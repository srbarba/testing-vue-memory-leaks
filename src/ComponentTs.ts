import Vue from 'vue'
import { mapGetters } from 'vuex'

export default Vue.extend({
  computed: {
    ...mapGetters(['userName']),
  },
  render(h) {
    return h('div', `Hello, ${this.userName}`)
  },
})
