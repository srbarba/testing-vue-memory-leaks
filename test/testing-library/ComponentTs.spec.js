import { renderWithVuetify } from '../renderWithVuetify'
import ComponentTs from '../../src/ComponentTs'

describe('ComponentTs', () => {
  it('renders correctly', () => {
    const wrapper = renderWithVuetify(ComponentTs)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders correctly with props', () => {
    const wrapper = renderWithVuetify(ComponentTs, {
      propsData: {
        text: 'My dummy Component',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
