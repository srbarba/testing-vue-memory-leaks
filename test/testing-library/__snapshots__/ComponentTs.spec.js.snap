// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentTs renders correctly 1`] = `<div class="v-card v-sheet theme--light">Dummy component</div>`;

exports[`ComponentTs renders correctly with props 1`] = `<div class="v-card v-sheet theme--light">My dummy Component</div>`;
